src/engine/SpatialJoinAlgorithms.cpp

```diff
@@ -50,60 +50,142 @@ std::string SpatialJoinAlgorithms::betweenQuotes(
   }
 }
 
-struct DistanceVisitor : public boost::static_visitor<double> {
-  template <typename Geometry1, typename Geometry2>
-  double operator()(const Geometry1& geom1, const Geometry2& geom2) const {
-    return bg::distance(geom1, geom2);
-  }
+bool SpatialJoinAlgorithms::getAnyGeometry(const IdTable* idtable, size_t row,
+                                           size_t col,
+                                           AnyGeometry& geometry) const {
```

Comment:
```suggestion
// __________________________________________________________________________________________________
std::optional<AnyGeometry> SpatialJoinAlgorithms::getAnyGeometry(const IdTable* idtable, size_t row,
                                           size_t col
                                         ) const {
```
Is the better interface here.